gatsby-node.js

```
@@ -195,23 +196,40 @@ exports.createPages = async ({ actions }) => {
     throw err;
   }
 
+  const openapiPageMetadata = manifestMetadata['openapi_pages'];
+  const openapiPageMapping = !!openapiPageMetadata ? await constructOpenAPIPageMapping(openapiPageMetadata) : {};
+
   return new Promise((resolve, reject) => {
     PAGES.forEach((page) => {
       const pageNodes = RESOLVED_REF_DOC_MAPPING[page]?.ast;
 
       const slug = getPageSlug(page);
       if (RESOLVED_REF_DOC_MAPPING[page] && Object.keys(RESOLVED_REF_DOC_MAPPING[page]).length > 0) {
+        const context = {
+          slug,
+          repoBranches,
+          associatedReposInfo,
+          isAssociatedProduct,
+          template: pageNodes?.options?.template,
+          page: pageNodes,
+        };
+
+        if (openapiPageMetadata[slug]) {
+          const openapiSpecStore = openapiPageMapping[slug];
+          if (!!openapiSpecStore) {
+            context['openapiSpecStore'] = openapiSpecStore;
+          } else {
+            // Skip content page and log error if an OpenAPI content page was
+            // unsuccessful sourcing its spec content.
+            console.error(`OpenAPI content page "${slug}" detected, but no spec store found.`);
+            return;
```

Most likely to be unsuccessful for `cloud-docs`/Atlas specs that require pulling the latest version via git hash. Sometimes, the git hash returned doesn't always have an existing OpenAPI spec, leading to a fetch error.

For now, we just skip building the page if there's a mismatch between an expected OpenAPI content page and what spec stores were created. Happy to change this if there's a stronger preference for the entire build failing. (I don't remember `mut`'s logic as of writing this, but I'm assuming that the stale/outdated page will still exist, even if skipped in the current build)

---

if this log will be captured, can we fall back to loading the page without the specs? otherwise might be safer to kill than a silent failure? my 2 cents

---

Rearranged [try-catch blocks](https://github.com/mongodb/snooty/pull/719/commits/40fc57d53631975ca2d4b3bfac5ce1d8c9a33453#diff-a6f3904c41211be170d9b0eb87e49ea8d02d0fd5a7224d6772174b425dacf90fR202-R209) to avoid silent failure for OpenAPI content pages